src/reactor/global.rs

```diff
@@ -49,6 +49,12 @@ impl Drop for HelperThread {
 
 fn run(mut reactor: Reactor, done: Arc<AtomicBool>) {
     while !done.load(Ordering::SeqCst) {
-        reactor.turn(None);
+        match reactor.turn(None) {
+            Err(_) => {
+                drop(reactor);
```

Comment:
Why is the drop needed here? Since it's owned by the run function is should be dropped at the end of the scope anyway.

Reply:
Ah, my bad, you're right. Fixed